bluemira/equilibria/run.py

```diff
@@ -104,21 +106,178 @@ def __post_init__(self):
 
 
-class PulsedCoilsetDesign:
+@dataclass
+class BreakdownCOPSettings:
+    """Breakdown settings for PulsedCoilsetDesign"""
+
+    B_stray_con_tol: float
+    n_B_stray_points: int
+
+
+@dataclass
+class EQSettings:
+    """Equilibrium settings for PulsedCoilsetDesign"""
+
+    coil_mesh_size: float
+    gamma: float
+    relaxation: float
+    peak_PF_current_factor: float
+
+
+@dataclass
+class PulsedCoilsetDesignFrame(ParameterFrame):
+    """PulsedCoilsetDesign Parameters"""
+
+    A: Parameter[float]
+    B_premag_stray_max: Parameter[float]
+    C_Ejima: Parameter[float]
+    I_p: Parameter[float]
+    l_i: Parameter[float]
+    R_0: Parameter[float]
+    tau_flattop: Parameter[float]
+    tk_sol_ib: Parameter[float]
+    v_burn: Parameter[float]
+
+
+class PulsedCoilsetDesign(ABC):
     """
     Abstract base class for the procedural design of a pulsed tokamak poloidal field
     coilset.
+
+    Parameters
+    ----------
+    params
+        Parameter frame with which to perform the problem
+    coilset
+        PF coilset to use in the equilibrium design
+    grid
+        Grid to use in the equilibrium design
+    equilibrium_constraints
+        List of magnetic constraints to use for equilibria. Depending on the optimisation
+        problem, these may be used in the objective function or constraints
+    profiles
+        Plasma profile object to use when solving equilibria
+    breakdown_strategy_cls
+        BreakdownZoneStrategy class to use when determining breakdown constraints
+    breakdown_problem_cls
+        Coilset optimisation problem class for the breakdown phase
+    breakdown_optimiser
+        Optimiser for the breakdown
+    breakdown_settings
+        Breakdown optimiser settings
+    equilibrium_problem_cls
+        Coilset optimisation problem class for the equilibria and current vector
+    equilibrium_optimiser
+        Optimiser for the equilibria and current vector
+    equilibrium_convergence
+        Convergence criteria to use when solving equilibria
+    equilibrium_settings
+        Settings for the solution of equilibria
+    current_opt_constraints
+        List of current optimisation constraints for equilibria
+    coil_constraints
+        List of coil current optimisation constraints for all snapshots (including
+        breakdown)
+    limiter
+        Limiter to use when solving equilibria
     """
 
     BREAKDOWN = "Breakdown"
     EQ_REF = "Reference"
     SOF = "SOF"
     EOF = "EOF"
 
-    def __init__(self):
+    def __init__(
+        self,
+        params,
+        coilset: CoilSet,
+        grid: Grid,
+        equilibrium_constraints: MagneticConstraintSet,
+        profiles: Profile,
+        breakdown_strategy_cls: Type[BreakdownZoneStrategy],
+        breakdown_problem_cls: Type[BreakdownCOP],
+        breakdown_optimiser: Optimiser = Optimiser(
+            "COBYLA", opt_conditions={"max_eval": 5000, "ftol_rel": 1e-10}
+        ),
```

Comment:
Because it is input to a function that is called on import of the module I dont think we can. The default argument will be an instance of Optimiser.

Reply:
From external conversation the Optimiser object itself is not immutable because python object are generally mutable therefore how these defaults are set up needs to change